### ISSUE TITLE
Bug in link rendering

### ISSUE THREAD

**Issue by iver56** [lionleaf/dwitter#334]
https://www.dwitter.net/d/435

That rendered comment looks buggy. I think the original comment just had a link there

![bilde](https://user-images.githubusercontent.com/1470603/39097575-73fa392e-465e-11e8-8a57-e8fab54a78c4.png)

**Comment by lionleaf** [comment]
Oh that's kinda nasty. Guess we have to make sure that any hashtag is not part of a valid URL... Hmm.

**Comment by stianjensen** [comment]
Another: https://www.dwitter.net/d/477

**Comment by joeytwiddle** [comment]
How about requiring a word boundary before the `#`? That is `\<#` in some regexp syntaxes, `\b#` in others.

Doing that would ignore the following:
```
abc#def
foo.com/bar#baz
```

But it would still match the following:
```
foo.com/bar.#baz
```

Or if you also wanted to ignore the latter, you could get more strict: require a blank space before the hashtag (or start of string). That would be `(^|\s)#` in some regexp syntaxes.

That would not even match this:
```
file this under:#banana
```

**Comment by stianjensen** [comment]
Do other platforms allow hashtags that are not preceded by a space?

**Comment by lionleaf** [comment]
Looks like it's fairly common to require a space or a special character before a hashtag, but varies per platform. Autocomplete and highlights as you write would solve this no matter which way we go. I guess a preceding space would make life easier, but slightly frustrating for users in some cases!#like_this

**Comment by lionleaf** [comment]
Another arguably false positive for hashtags: https://www.dwitter.net/d/1486

Thanks to fujimotos from discord

**Comment by atesgoral** [comment]
Yeah, I just ran into another scenario where a hex color code inside a code snippet was hyperlinked as a tag: `"#362":"#bb3"`

https://www.dwitter.net/d/9083

**Comment by kipkat** [comment]
I suggest something special for colors: [[hexcolor]].
Javascript version: (I don't know the django version)
```js
comment = `Yeah, I just ran into another scenario where a hex color code inside a code snippet was hyperlinked as a tag: "[[362]]":"[[bb3]]"`
// ALL THE OTHER FILTERS HERE
var newcomment = []
spl=comment.split("[[")
for(i=0;i<spl.length;i++)
{
    if(spl[i].indexOf("]]")>-1)
    {
	spl[i] = "#"+spl[i].replace("]]","")
    }
    // spl[i] is now the color. process it here
    newcomment.push(spl[i])
}

newcomment = newcomment.join("")
comment = newcomment
```
and now comment is `Yeah, I just ran into another scenario where a hex color code inside a code snippet was hyperlinked as a tag: "#362":"#bb3"`

Probably a on hovering the hexcolors -> change font color to specified color is a good idea too.

**Comment by lionleaf** [comment]
Here's another example from @yonatan: https://www.dwitter.net/d/2667  

@dwitterer has started looking at regex solutions.
